wire-ios-sync-engine/Source/UnauthenticatedSession/ZMAuthenticationStatus.swift

```diff
@@ -19,3 +19,24 @@
 import Foundation
 
 extension ZMAuthenticationStatus: NotificationContext {}
+
+extension ZMAuthenticationStatus: ZMCredentialProvider {
+
+    public func credentialsMayBeCleared() {
+        if self.currentPhase == ZMAuthenticationPhase.authenticated {
+            self.resetLoginAndRegistrationStatus()
+        }
+    }
+
+    public func emailCredentials() -> UserEmailCredentials! {
+        if let loginCredentials = self.loginCredentials, loginCredentials.credentialWithEmail {
+            return UserEmailCredentials.credentials(
+                email: loginCredentials.email ?? "",
+                password: loginCredentials.password ?? "",
+                emailVerificationCode: loginCredentials.emailVerificationCode
+            )
+        }
+        return nil
```

Comment:
we return nil but this is explicitly unwrapped : `public func emailCredentials() -> UserEmailCredentials!`

Reply:
Addressed it here: https://github.com/wireapp/wire-ios/commit/e93fc4bda3cdc961678f81f7980856357a3d3ef3